src/npm.py

```diff
@@ -0,0 +1,70 @@
+from glob import glob
+import os
+import re
+import sys
+import json
+import frontmatter
+import urllib.request
+
+DEFAULT_TAG_TEMPLATE = (
+    "{{major}}{% if minor %}.{{minor}}{% if patch %}.{{patch}}{%endif%}{%endif%}"
+)
+REGEX = r"^(?:(\d+\.(?:\d+\.)*\d+))$"
+
+
+def fetch_releases(npm_id, regex):
+    releases = {}
+
+    if not isinstance(regex, list):
+        regex = [regex]
+
+    url = f"https://registry.npmjs.org/{npm_id}"
+    with urllib.request.urlopen(url, data=None, timeout=5) as response:
+        data = json.loads(response.read().decode("utf-8"))
+        for version in data["time"]:
+            R = data["time"][version]
+            matches = False
+            for r in regex:
+                if re.match(r, version):
+                    matches = True
+            if matches and R:
+                d = ""
+                for c in R:
+                    if c == 'T':
+                        break
+                    d += c
+                releases[version] = d
+                print(f"{version}: {d}")
+
+    return releases
+
+
+def update_releases(product_filter=None):
+    for product_file in glob("website/products/*.md"):
+        product_name = os.path.splitext(os.path.basename(product_file))[0]
+        if product_filter and product_name != product_filter:
+            continue
+        with open(product_file, "r") as f:
+            data = frontmatter.load(f)
+            if "auto" in data:
+                for config in data["auto"]:
+                    for key, d_id in config.items():
+                        if key == "npm":
+                            update_product(product_name, config)
+
+
+def update_product(product_name, config):
+    if "npm" in config:
+        print(f"::group::{product_name}")
+        config = config | {"regex": REGEX}
```

Comment:
Think there is a problem with this merge: regex will always be `REGEX` even if it is provided in the config because the second dict overrides the first one.

See:
```
$ python3
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> dict1 = {'regex': 'abc'}
>>> dict2 = {'regex': 'def'}
>>> print(dict1 | dict2)
{'regex': 'def'}
```

Reply:
Fixed for both npm and pypi.